### PR TITLE
Phar build fix

### PR DESCRIPTION
**Pull request by dantleech**
Fix file path handling to make Phpactor compatible with a PHAR distribution.

This PR also introduces a development `docker-compose.yml` to allow the full tooling stack to be executed locally without fuss.

### CODE REVIEW

File: lib/TextDocument/Tests/Unit/TextDocumentUriTest.php [hunk 1]
```diff
@@ -14,12 +14,17 @@ public function testCreate(): void
...
-    // TODO: This doesn't work
-    //public function testCreateUntitled(): void
-    //{
-    //    $uri = TextDocumentUri::fromString('file://' . __FILE__);
-    //    $this->assertEquals('file://' . __FILE__, (string) $uri);
-    //}
+    public function testCreateUntitledOtherScheme(): void
+    {
+        $uri = TextDocumentUri::fromString('untitled:Untitled-1');
+        $this->assertEquals('untitled:Untitled-1', (string) $uri);
```
Comment (dantleech) [inline lib/TextDocument/Tests/Unit/TextDocumentUriTest.php:20]: rename

File: lib/Filesystem/Domain/FilePath.php [hunk 2]
```diff
@@ -88,12 +70,12 @@ public static function fromUnknown($path): FilePath
...
     public function isDirectory(): bool
     {
-        return is_dir($this->path);
+        return is_dir($this->uri->path());
     }
 
     public function asSplFileInfo(): SplFileInfo
     {
-        return new SplFileInfo($this->path());
+        return new SplFileInfo($this->uri->path());
```
Comment (dantleech) [inline lib/Filesystem/Domain/FilePath.php:78]: SplFileInfo is probably happy taking the scheme

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -12,6 +12,44 @@ env:
...
 
 jobs:
+    phar:
+        name: Compile Phar
+        runs-on: ubuntu-latest
+        steps:
+            - name: Checkout code
+              uses: actions/checkout@v3
+              with:
+                  fetch-depth: 0
```
Comment (drupol) [inline .github/workflows/ci.yml:22]: Is this really needed ? This will download the whole git history... while only the latest commit is relevant here.

Comment (dantleech) [reply .github/workflows/ci.yml:22]: thanks, copy/paste from another place, fixed in `master`

Comment (drupol) [reply .github/workflows/ci.yml:22]: Cool, thanks :)

**Pull request merged**